src/Xamarin.Android.Build.Tasks/Tasks/Generator.cs

```diff
@@ -133,7 +139,20 @@ public override bool RunTask ()
 			if (Log.HasLoggedErrors)
 				return false;
 
-			return base.RunTask ();
+			var result = base.RunTask ();
+			List<ITaskItem> files = new List<ITaskItem> ();
+			if (result && GeneratedFileListFile != null && File.Exists (GeneratedFileListFile)) {
+				var doc = XDocument.Load (GeneratedFileListFile);
+				var compileItems = doc.XPathSelectElements ("//Project/ItemGroup/Compile");
```

Comment:
I'm surprised that this works, given that the `generator`-generated `.projitems` file uses XML Namespaces: 

 
 * https://github.com/xamarin/java.interop/blob/651de42732d194cee5a45fae45feda37706a8c16/tools/generator/GenerationInfo.cs#L40-L68
  * https://github.com/xamarin/java.interop/blob/651de42732d194cee5a45fae45feda37706a8c16/tests/generator-Tests/expected.ji/TestInterface/Mono.Android.projitems

Wouldn't `compileItems` *always* be an empty iterator right now?

Reply:
It worked in my tests?

Reply:
Not sure why it worked in the tests; if I apply this patch:

```diff
diff --git a/src/Xamarin.Android.Build.Tasks/Tasks/Generator.cs b/src/Xamarin.Android.Build.Tasks/Tasks/Generator.cs
index a3ee0e211..7b58ed3f4 100644
--- a/src/Xamarin.Android.Build.Tasks/Tasks/Generator.cs
+++ b/src/Xamarin.Android.Build.Tasks/Tasks/Generator.cs
@@ -142,11 +142,13 @@ namespace Xamarin.Android.Tasks
 			var result = base.RunTask ();
 			List<ITaskItem> files = new List<ITaskItem> ();
 			if (result && GeneratedFileListFile != null && File.Exists (GeneratedFileListFile)) {
+				Log.LogDebugMessage ($"# jonp: SHOULD process {GeneratedFileListFile}");
 				var doc = XDocument.Load (GeneratedFileListFile);
 				var compileItems = doc.XPathSelectElements ("//Project/ItemGroup/Compile");
 				foreach (var item in compileItems) {
 					var file = item.Attribute ("Include");
 					if (file != null && File.Exists (file.Value)) {
+						Log.LogDebugMessage ($"# jonp: found //Compile/@Include value: {file.Value}");
 						files.Add (new TaskItem (file.Value));
 					}
 				}
```

then build this branch locally, and do the equivalent of:

```sh
% dotnet new androidlib
% cat > Example.java <<EOF
package e;

public class Example {
    public static void e() {
    }
}
EOF
% dotnet build -v:diag > b.txt
```

Then I see:

```
# jonp: SHOULD process obj/Debug/net9.0-android/generated/src/gxa-8706.projitems
```

but I *don't* see `# jonp: found //Compile/@Include value`.  I also believe that the `GeneratedFiles` output property is empty, as it doesn't appear in the diagnostic log.

The contents of `gxa-8706.projitems`:

```xml
<?xml version="1.0" encoding="utf-8"?>
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <PropertyGroup>
    <DefineConstants>$(DefineConstants);ANDROID_1;ANDROID_2;ANDROID_3;ANDROID_4;ANDROID_5;ANDROID_6;ANDROID_7;ANDROID_8;ANDROID_9;ANDROID_10;ANDROID_11;ANDROID_12;ANDROID_13;ANDROID_14;ANDROID_15;ANDROID_16;ANDROID_17;ANDROID_18;ANDROID_19;ANDROID_20;ANDROID_21;ANDROID_22;ANDROID_23;ANDROID_24;ANDROID_25;ANDROID_26;ANDROID_27;ANDROID_28;ANDROID_29;ANDROID_30;ANDROID_31;ANDROID_32;ANDROID_33;ANDROID_34</DefineConstants>
  </PropertyGroup>
  <!-- Classes -->
  <ItemGroup>
    <Compile Include="$(MSBuildThisFileDirectory)E.Example.cs" />
    <Compile Include="$(MSBuildThisFileDirectory)Java.Interop.__TypeRegistrations.cs" />
    <Compile Include="$(MSBuildThisFileDirectory)__NamespaceMapping__.cs" />
  </ItemGroup>
  <!-- Enums -->
  <ItemGroup />
</Project>
```

Reply:
…so I use [this stackoverflow answer](https://stackoverflow.com/a/19866701) to add use of an `XmlNamespaceManager` so that the XPath query can. use namespaces, resulting in:

```csharp
		void AddGeneratedFiles ()
		{
			List<ITaskItem> files = new List<ITaskItem> ();
			Log.LogDebugMessage ($"# jonp: SHOULD process {GeneratedFileListFile}");
			var doc = XDocument.Load (GeneratedFileListFile);

			var nsmgr = new XmlNamespaceManager(new NameTable());
			nsmgr.AddNamespace ("msb", "http://schemas.microsoft.com/developer/msbuild/2003");

			var compileItems = doc.XPathSelectElements ("//msb:Project/msb:ItemGroup/msb:Compile", nsmgr);
			foreach (var item in compileItems) {
				var file = item.Attribute ("Include");
				if (file != null && File.Exists (file.Value)) {
					Log.LogDebugMessage ($"# jonp: found //Compile/@Include value: {file.Value}");
					files.Add (new TaskItem (file.Value));
				}
			}
			GeneratedFiles = files.ToArray ();
		}
```

and realize, to my horror, that it *still* doesn't work, because of the `File.Exists()` check: the value of `file.Attribute("Include").Value` will be `$(MSBuildThisFileDirectory)E.Example.cs`, ***which will not exist***.  It will (almost certainly) *never* exist, because a directory literally named `$(MSBuildThisFileDirectory)` will (almost certainly) never exist.  (Unless you're bananas and do `mkdir '$(MSBuildThisFileDirectory)'`, but who would do such a thing?)

This in turn means that, *as-is*, `@(_GeneratedBindingFiles)` will *always* be empty, which suggests that most of the changes to the `GenerateBindings` target are in fact dead code.

Reply:
…which is a long-winded way of explaining that I think this *doesn't* work, and is effectively dead (unexecuted) code.

I'm trying to remove it; let's see if anything breaks.